src/Jackalope/Node.php

```diff
@@ -1656,8 +1683,9 @@ public function beginTransaction()
         parent::beginTransaction();
 
         // Notify the children properties
-        foreach ($this->properties as $prop) {
-            $prop->beginTransaction();
+        foreach ($this->propertyValues as $name => $propertyValue) {
+            $property = $this->loadProperty($name);
```

Comment:
and again here, non existing properties don't need to know about this, i think. or do they?